pvlib/iotools/crn.py

```diff
@@ -103,19 +110,13 @@ def read_crn(filename):
     dtindex = pd.to_datetime(dts['UTC_DATE'] + dts['UTC_TIME'].str.zfill(4),
                              format='%Y%m%d%H%M', utc=True)
     data = data.set_index(dtindex)
-    try:
-        # to_datetime(utc=True) does not work in older versions of pandas
-        data = data.tz_localize('UTC')
-    except TypeError:
-        pass
 
     # Now we can set nans. This could be done a per column basis to be
     # safer, since in principle a real -99 value could occur in a -9999
     # column. Very unlikely to see that in the real world.
-    for val in [-99, -999, -9999]:
-        # consider replacing with .replace([-99, -999, -9999])
-        data = data.where(data != val, np.nan)
+    data = data.replace([-99, -999, -9999, -99999, -999999], np.nan)
```

Comment:
After further review of the [documentation](https://www.ncei.noaa.gov/pub/data/uscrn/products/subhourly01/) and file format, I believe -999999 is not possible because all of the 7 character data fields contain at least one decimal point.

Reply:
Duly noted.